subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/recomp/GroovyRecompilationSpecProvider.java

```diff
@@ -110,6 +111,11 @@ private void processOtherChanges(PreviousCompilation previous, RecompilationSpec
             }
 
             File changedFile = fileChange.getFile();
+            if (FileUtils.hasExtension(changedFile, ".java")) {
+                spec.setFullRebuildCause("Groovy-Java joint compilation detected", changedFile);
```

Comment:
How about: `Changes to non-Groovy files are not supported for incremental compilation`?

Reply:
Can't capitalize the sentense here, the final log is `Full recompilation is required because XXXX`.

Changed to `changes to non-Groovy files are not supported by incremental compilation`.